adit/dicom_web/utils/wadors_utils.py

```
@@ -95,3 +113,187 @@ async def add_sentinel_when_done(task: asyncio.Task[None]) -> None:
+
+                async for filename, file_content in _process_single_fetch(dicom_images):
+                    yield filename, file_content
+        else:
+            dicom_images = await sync_to_async(
+                _fetch_dicom_data, thread_sensitive=False
+            )(source_server, query, level)
+            async for filename, file_content in _process_single_fetch(dicom_images):
+                yield filename, file_content
+
+    except RetriableDicomError as err:
+        raise ServiceUnavailableApiError(str(err))
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)